actions/contract_tests/goalieruns.js

```diff
@@ -10,62 +11,53 @@ module.exports.runTests = async function runTests() {
-          // if (err) {
-          //   try {
-          //     // send this to the right slack channel
-          //     // the start and end indices, as well as the length, are constant because the stacktrace always
-          //     // includes the same spots in the tape module (only the failing test itself may change)
-          //     const startIndex = stdout.indexOf(
-          //       "at Test.bound [as equal] (/github/workspace/node_modules/tape/lib/test.js:91:32)"
-          //     );
-          //     const endIndex = stdout.indexOf(
-          //       "at processTicksAndRejections (internal/process/task_queues.js:93:5)"
-          //     );
-          //     const len = "at Test.bound [as equal] (/github/workspace/node_modules/tape/lib/test.js:91:32)"
-          //       .length;
-          //     const testMessage = stdout.slice(startIndex + len, endIndex);
-          //     const expectedCode = stdout.slice(
-          //       stdout.indexOf("expected: ") + 10,
-          //       stdout.indexOf("      actual:")
-          //     );
-          //     const actualCode = stdout.slice(
-          //       stdout.indexOf("actual:   ") + 10,
-          //       stdout.indexOf("      at:")
-          //     );
-          //     const block = [
-          //       `_FAILED CONTRACT TEST_:      ${testMessage}`,
-          //       `_EXPECTED_:  ${expectedCode}`,
-          //       `_ACTUAL_:      ${actualCode}`,
-          //     ];
-          //     const result = await web.chat.postMessage({
-          //       channel: pkg.config.goalieMappings[validated_arg].slackChannel,
-          //       text: `:sadpanda: ${block.join("\n")}`,
-          //     });
-          //   } catch (error) {
-          //     console.error(error);
-          //     return error.code;
-          //   }
-          // }
           if (err) {
-            console.log(err);
-            return err.code;
+            const startIndex = stdout.indexOf("✖");
+            const endIndex = stdout.indexOf("stack:");
+            failures.push(stdout.slice(startIndex, endIndex));
+          }
+          if (++count == test_set.length) {
+            console.log("REACHED END, RESOLVING");
+            console.log("VALUE OF COUNT: ", count);
+            return resolve();
           }
           console.log(stdout);
         });
       }
-    } catch (joiError) {
-      console.error(JSON.stringify(joiError, null, 2));
-      return joiError.code;
-    }
+    }).then(async function () {
+      if (failures.length > 0) {
+        let errorMessage = "";
+        if (failures.length > 1) {
+          errorMessage = `There were ${failures.length} failures in the contract tests:`;
+        } else {
+          errorMessage = `There was 1 failure in the contract tests:`;
+        }
+        try {
+          const result = await web.chat.postMessage({
+            channel: "more-tests",
+            text: `:sadpanda: ${errorMessage}
+  ${failures.join("\n")}`,
```

Comment:
I'm a big fan of Slack's Block Kit library. It might be useful in a future commit to leverage some formatting and layout capabilities to improve the readability of the generated error message. I found the error messages readable on desktop, but the notification alert that appeared on my iPhone could have been titled better by taking advantage of Block Kit's formatting options.